JS/index.js

```diff
@@ -0,0 +1,184 @@
+        const flagImage = cloneCard.querySelector(".country-flag img");
+        const countryInfo = cloneCard.querySelectorAll(".country-brief li");
+
+        titleElement.innerHTML = country.name;
+        flagImage.src = country.flag;
+        flagImage.alt = `${country.name}`;
+        
+        countryInfo[0].innerHTML = `<strong>Population:</strong> ${country.population}`;
+        countryInfo[1].innerHTML = `<strong>Region:</strong> ${country.region}`;
+        countryInfo[2].innerHTML = `<strong>Capital:</strong> ${country.capital}`;
+
+        containerCountries.appendChild(cloneCard);
+    });
+};
+
+const openDropDown = () => {
+  const dropDown = document.querySelector(".dropdown-wrapper");
+  console.log('enter');
+  dropDown.classList.toggle("open");
+}
+
+const filterByRegion = (event) => {
+   const region = event.target.getAttribute("data-region");
+   console.log("select region:" + region)
+
+   if(region!="all"){
+   const result = countries.filter((country)=>{
+    return country.region.toLowerCase() == region.toLowerCase()
+   })
+   console.log(result);
+   innerDetailsIntoCards(result);
+}
+else{
+    innerDetailsIntoCards(countries);
+}
+
+//    const result =  countries.filter((country)=>{
+//          return country.region === "Africa";
+//     })
+//     console.log(result);
+}
+
+const matchValues = (country, input) => {
+    return (country.name.toLowerCase().includes(input) || 
+        country.population.toString().includes(input)) ||
+        country.region.toLowerCase().includes(input) ||
+        country.capital.toLowerCase().includes(input)
+}
+const filterBySerach = (event) =>{
+    console.log("enter to filterbyserach func");
+    const input = event.target.value.toLowerCase();
+    console.log(input);
+
+    const resultBySearch = countries.filter((country)=>{
+        return matchValues(country,input);
+     })
+    console.log(resultBySearch);
+    innerDetailsIntoCards(resultBySearch);
+}
+
+
+const switchPage = (event) =>{
+    const card = event.currentTarget;
+    console.log(card.innerHTML);
+    
+    const titleElement = card.querySelector(".country-title").textContent;
+    const flagImage = card.querySelector(".country-flag img").src;
+    const population = card.querySelector(".country-brief li:nth-child(1)").textContent;
+    const region = card.querySelector(".country-brief li:nth-child(2)").textContent;
+    const capital = card.querySelector(".country-brief li:nth-child(3)").textContent;
+    
+    const countryInfo = {name:titleElement, flagImg:flagImage, population:population, region: region, capital:capital};
+    console.log(countryInfo);
+    localStorage.setItem("selectedCountry", JSON.stringify(countryInfo));
+    window.location.href = "details.html";
+}
+
+
+//maybe do this
+const changeColorOfPage = () =>{
+    console.log("enter");
+    const currentColor = getComputedStyle(document.body).backgroundColor;
+    
+    if (currentColor === "rgb(33, 37, 41)") { 
```

Comment:
consider to assign the hard-coded values: `"rgb(33, 37, 41)"` and `"rgb(255, 255, 255)"` to constant variables.

consider to create a new file for your application constants: `const.js`